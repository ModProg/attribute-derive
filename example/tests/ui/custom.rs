use example::Custom;

#[derive(Custom)]
struct None;

#[derive(Custom)]
#[ident(example = 2.)]
struct Example;

#[derive(Custom)]
#[ident(
    example = 2.,
    optional_implicit = {10},
    optional_default = 3,
    default = 2,
    conflict_a = "hello",
)]
struct ExampleOI;

#[derive(Custom)]
#[ident(example = 1.)]
#[a(conflict_a = "hey")]
#[b(conflict_b = "hi")]
struct Conflict;

#[derive(Custom)]
#[ident(hello)]
#[single(hello)]
#[empty(hello)]
struct UnknownField;

fn main() {}
