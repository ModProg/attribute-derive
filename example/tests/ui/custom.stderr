error: missing field `example`

       = help: try ident: example=2.5
 --> tests/ui/custom.rs:3:10
  |
3 | #[derive(Custom)]
  |          ^^^^^^
  |
  = note: this error originates in the derive macro `Custom` (in Nightly builds, run with -Z macro-backtrace for more info)

error: missing flag `mandatory_flag`

       = help: try ident: mandatory_flag
 --> tests/ui/custom.rs:6:10
  |
6 | #[derive(Custom)]
  |          ^^^^^^
  |
  = note: this error originates in the derive macro `Custom` (in Nightly builds, run with -Z macro-backtrace for more info)

error: conflict_a !!! conflict_b
  --> tests/ui/custom.rs:23:5
   |
23 | #[a(conflict_a = "hey")]
   |     ^^^^^^^^^^

error: conflict_b !!! conflict_a
  --> tests/ui/custom.rs:24:5
   |
24 | #[b(conflict_b = "hi")]
   |     ^^^^^^^^^^

error: found `hello` but expected one of `optional_implicit`, `optional_explicit`, `optional_default`, `default`, `conflict_a`, `conflict_b`, `example`, `flag`, `mandatory_flag`
  --> tests/ui/custom.rs:28:9
   |
28 | #[ident(hello)]
   |         ^^^^^

error: found hello, but expected none
  --> tests/ui/custom.rs:30:9
   |
30 | #[empty(hello)]
   |         ^^^^^

error: found hello, but expected field
  --> tests/ui/custom.rs:29:10
   |
29 | #[single(hello)]
   |          ^^^^^
